### PR TITLE
INTLY-7424 Verify RHMI products status

### PR DESCRIPTION
**Pull request by sergioifg94**
# Description

> Link to JIRA: https://issues.redhat.com/browse/INTLY-7424

Extend tests for RHMI stages completion (#703) with verification of the products status within each stage.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [ ] This change requires a documentation update 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Verified independently on a cluster by reviewer

### CODE REVIEW

**Comment by CathalOConnorRH** [comment]
Tested and verified. 
/lgtm

File: test/common/integreatly_stages.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+		rhmi, err := getRHMI(ctx.Client)
+		if err != nil {
+			return false, fmt.Errorf("error getting RHMI CR: %v", err)
+		}
+
+		//iterate stages and check their status
+		for stageName, productNames := range expectedStageProducts {
+			stage, ok := rhmi.Status.Stages[v1alpha1.StageName(stageName)]
+			if !ok {
+				t.Errorf("Error checking stage %s. Not found", stageName)
```
Comment (JameelB) [inline test/common/integreatly_stages.go:60]: Should it not exit out of the poll as soon as it gets an error? You never actually return an error here for this line to be called: https://github.com/integr8ly/integreatly-operator/pull/707/files#diff-0db09d831d48ec632033bf219619b029R99

What's the reason for continuing the test once it fails here?

Comment (sergioifg94) [reply test/common/integreatly_stages.go:60]: I don't exit the poll immediately in order to continue testing for other possible failed products or stages. However I set the `done` variable to `true`, so the poll will finish after testing all the stages

**Comment by sergioifg94** [comment]
@JameelB I think this all comes from the fact that there's two separate jiras for them, so @CathalOConnorRH  and myself were working on them separately until he noticed that they're related, then we decided to incorporate my work in his test.

**Comment by CathalOConnorRH** [comment]
@JameelB I'm happy to close my PR and update the jira to link to PR @sergioifg94 is doing.

**Comment by JameelB** [comment]
@CathalOConnorRH I think that makes more sense, no point in getting the other one in if it's going to be changed in this PR anyway. @sergioifg94 this can be the PR linked for both INTLY-7423 AND intly-7424

thanks @CathalOConnorRH!

File: test/common/integreatly_stages.go (hunk 1 above)
Comment (JameelB) [reply test/common/integreatly_stages.go:60]: Explained - Will continue the tests here to report all errors for all stages and products.

**Pull request merged**